editor/components/post-permalink/editor.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { withDispatch } from '@wordpress/data';
+import { Component, compose } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+import { getEditedPostAttribute } from '../../store/selectors';
+
+class PostPermalinkEditor extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			editedPostName: '',
+		};
+
+		this.onSavePermalink = this.onSavePermalink.bind( this );
+	}
+
+	/**
+	 * Returns the permalink parts to populate the form.
+	 *
+	 * @return {Object} The prefix, suffix, and postName for the form.
+	 */
+	getPermalinkParts() {
+		const [ template, postName ] = this.props.samplePermalinkData;
+		const [ samplePermalinkPrefix, samplePermalinkSuffix ] = template.split( /%(?:postname|pagename)%/ );
+
+		return {
+			samplePermalinkPrefix,
+			samplePermalinkSuffix,
+			postName,
+		};
+	}
+
+	onSavePermalink() {
+		const postName = this.state.editedPostName.replace( /\s+/g, '-' );
+		const [ template, oldPostName ] = this.props.samplePermalinkData;
+
+		this.props.onSave();
+
+		if ( ! postName || postName === oldPostName ) {
+			return;
+		}
+
+		this.props.editPost( {
+			slug: postName,
+			sample_permalink: [ template, postName ],
+		} );
+
+		this.setState( {
+			editedPostName: postName,
+		} );
+	}
+
+	render() {
+		const {
+			samplePermalinkPrefix,
+			samplePermalinkSuffix,
+			postName,
+		} = this.getPermalinkParts();
+
+		/* eslint-disable jsx-a11y/no-autofocus */
```

Comment:
I tend to encourage that whenever we disable a rule, we include some clearly explained rationale that explains why it's okay to be disabled. Here, something like as described at https://github.com/WordPress/gutenberg/pull/5756#issuecomment-376127229 would suffix (transient UI for editing). It's also a good sanity check for oneself.

Reply:
👍🏻